### DISCUSSION TITLE
Objets Active Directory

### DISCUSSION THREAD

**Discussion by mnotin** [dbarzin/mercator#731] [Q&A]
Bonjour,

Est-il prévu d'ajouter la possibilité de déclarer des objets (notamment des utilisateurs) rattachés à un annuaire au niveau de la vue de l'administration ?

Cordialement

**Comment by dbarzin** [comment]
Cela n'a jamais été évoqué. 
A quoi penses-tu ? Ajouter un objet utilisateur attaché à un annuaire ? 
Que souhaites-tu en faire ? La revue des droits d'accès ?

**Comment by mnotin** [comment]
Oui, l'idée serait de pouvoir ajouter un objet utilisateur. Ce dernier serait rattaché à un annuaire si ce n'est pas un compte local à une machine. L'objectif serait de pouvoir se mettre en conformité avec la [directive NIS](https://cyber.gouv.fr/la-directive-nis) (dont les règles de sécurité sont définies dans [l'arrêté du 14 septembre 2018](https://www.legifrance.gouv.fr/affichTexte.do?cidTexte=JORFTEXT000037444012&dateTexte=&categorieLien=id)) et donc ici de constituer une liste des comptes disposant de droits d'accès privilégiés au système d'information (appelés «comptes privilégiés»). Cette liste préciserait pour chaque compte le niveau de droits et le type de compte (comptes utilisateur, compte de messagerie, compte de processus, etc).

Les différents attributs d'un objet utilisateur pourraient donc être :
- Nom
- Prénom
- Compte local (oui / non)
- Compte privilégié (oui / non)
- Type (compte utilisateur, compte de messagerie, compte de processus, etc)
- Description

**Comment by dbarzin** [comment]
Du point de vue de la modélisation des données, tu vois cela comment ?
Je propose ceci : un utilisateur est lié (facultativement) à un domaine.
On peut ajouter un champ login ou userid (obligatoire) qui est unique (par domaine ?)
Les champs nom, prénom sont facultatifs si c'est un compte de service.
Comment modéliser les comptes locaux, à quels objets sont-ils liés ?

**Comment by mnotin** [comment]
La modélisation des données me semble parfaite comme cela.
Un userid devrait être unique par domaine oui.
Et pour les comptes locaux, je pense que l'on pourrait les lier à un objet poste de travail.

**Comment by dbarzin** [comment]
On se limite aux postes de travail pour les administrateurs locaux ?

**Comment by mnotin** [comment]
En effet, il pourrait être intéressant de pouvoir associer un utilisateur local à d'autres types d'équipements.
Je pense notamment à un commutateur logique, un routeur logique, une équipement de sécurité logique, un serveur DHCP, un serveur DNS et un serveur logique.

**Comment by dbarzin** [comment]
Cela va poser de difficultés de modélisation, surtout si tous ces liens sont tous n-m. Dans un premier temps, je propose de limiter le lien entre utilisateurs et domaines. Le champ description pourra servir à décrire le détail des droits accordés.

**Comment by mnotin** [comment]
Très bien, ce sera parfait dans un premier temps. Merci beaucoup pour tout le travail déjà réalisé sur la fonctionnalité en tout cas.
Serait-il possible de pouvoir aussi associer un serveur logique avec un domaine Active Directory s'il vous plaît ?

**Comment by dbarzin** [comment]
C'est dispo !
